app/helpers/datasets_helper.rb

```diff
@@ -51,68 +51,28 @@ def expected_location_class_for(dataset)
     "dgu-unavailable" if locations(dataset).empty?
   end
 
-  def timeseries_data?(datafiles)
-    datafiles_start_dates(datafiles).uniq.size > 1
-  end
-
-  def group_by_year(datasets)
-    datasets_with_year = datasets.map do |dataset|
-      dataset['start_year'] =
-          dataset['start_date'] ? Time.parse(dataset['start_date']).year.to_s : ''
-      dataset
-    end
-    Hash[datasets_with_year.group_by {|dataset| dataset['start_year']}.sort.reverse]
-  end
-
-  def link_type(datafile)
-    if datafile["format"].blank?
-      :no_preview
-    elsif datafile["format"].upcase == 'CSV'
-      :csv
-    else
-      datafile["format"].upcase == 'HTML' ? :html : :no_preview
-    end
-  end
-
   def name_of(dataset)
     dataset._source['name']
   end
 
   private
 
-  def datafiles_start_dates(datafiles)
-    datafiles_with_start_dates(datafiles).map { |file| Time.parse(file['start_date']).year }
-  end
-
-  def datafiles_with_start_dates(datafiles)
-    datafiles.select { |file| file['start_date'].present? }
-  end
-
   def datafile_next_updated(dataset)
     freq = dataset.frequency
-    last = most_recent_date(dataset.datafiles)
+    last = Time.parse(most_recent_date(dataset.datafiles))
```

Comment:
Calling `.advance` on `last` on the next line throws an error if the `most_recent_date` is parsed.

Reply:
Which is what the code was already doing. Good spot! I'll ship a fix.

Reply:
Hm, actually just tried in the Rails console, it works.

```
irb(main):004:0> Time.parse("1/1/15").advance(years: 1)
=> 2002-01-15 00:00:00 +0000
```

Reply:
I guess it's because

```
irb(main):005:0> Time.parse("1/1/15").class
=> Time
```

Never used `advance` before, cool little method.